### ISSUE TITLE
Issue with `AsAsyncValueEnumerable().SingleAsync`

### ISSUE THREAD

**Issue by Mike-E-angelo** [NetFabric/NetFabric.Hyperlinq#375]
Please see the tests here:
https://github.com/Mike-E-angelo/Stash/blob/master/Hyperlinq.SingleAsync/Hyperlinq.SingleAsync/SingleTests.cs#L13-L32

In it, I have `Hyperlinq` and `System.Linq.Async` polling an `IAsyncEnumerable` created from an `EF.CompileQueryAsync` call with their respective `SingleAsync` method.

`System.Linq.Async` passes while `Hyperlinq` does not, even though `Hyperlinq` reports that `Option<T>.IsSome` is `true`.

This seems wrong to me. 😁

**Comment by aalmada** [comment]
Strange! All my tests are passing...🧐
![image](https://user-images.githubusercontent.com/534533/131233192-61e0073a-a735-498e-b64f-d501c575ae71.png)
Maybe I'm not checking it correctly.😰

**Comment by Mike-E-angelo** [comment]
FWIW EFCore has thrown me for a couple loops in my world, too. 😅

**Comment by aalmada** [comment]
I'm trying to compile your test to reproduce the behavior. I think I have the same packages and namespaces but I'm getting a compile error. It says there's an ambiguous reference for the `Where` method used in the compiled query.

```
[CS0121] 
The call is ambiguous between the following methods or properties:
'System.Linq.AsyncEnumerable.Where<TSource>(System.Collections.Generic.IAsyncEnumerable<TSource>, System.Func<TSource, bool>)' 
and 
'System.Linq.Queryable.Where<TSource>(System.Linq.IQueryable<TSource>, System.Linq.Expressions.Expression<System.Func<TSource, bool>>)'
```

It's strange. Do you know why this is happening?

**Comment by Mike-E-angelo** [comment]
Wow that _is_ strange.  `dotnet clean` and `dotnet restore --force` are my usual go-tos for such oddities.

Are you compiling the project as-is or are you copy/pasting/transplanting it into a new project?

**Comment by aalmada** [comment]
I'm copy/pasting into my tests project

**Comment by aalmada** [comment]
I see that `DbSet<TEntity>` implements both `IQueriable<TEntity>` and `IAsyncEnumerable<TEntity>`. I now question how it didn't cause problems before.🧐

**Comment by aalmada** [comment]
So, long story short, somewhere in time I refactored `SingleAsync()` to something really stupid. The unit tests didn't catch it because the test returned `ValueTask` but looks like xUnit awaits only if the method returns `Task`. That means exceptions were simply ignored. 😱

I fixed the return type for all asynchronous tests and found a couple more failing tests. Unfortunately, the assertions on one of them have very strange behavior and I'm still trying to understand what's going on.
